src/Renci.SshNet/Security/Cryptography/Ciphers/TripleDesCipher.cs

```diff
@@ -1,154 +1,86 @@
 ﻿using System;
+using System.Security.Cryptography;
+
+#if !NET
+using Org.BouncyCastle.Crypto.Paddings;
+
+using Renci.SshNet.Security.Cryptography.Ciphers.Modes;
+#endif
 
 namespace Renci.SshNet.Security.Cryptography.Ciphers
 {
     /// <summary>
     /// Implements 3DES cipher algorithm.
     /// </summary>
-    public sealed class TripleDesCipher : DesCipher
+    public sealed partial class TripleDesCipher : BlockCipher, IDisposable
     {
-        private int[] _encryptionKey1;
-        private int[] _encryptionKey2;
-        private int[] _encryptionKey3;
-        private int[] _decryptionKey1;
-        private int[] _decryptionKey2;
-        private int[] _decryptionKey3;
+#if NET
+        private readonly BclImpl _impl;
+#else
+        private readonly BlockCipher _impl;
+#endif
 
         /// <summary>
         /// Initializes a new instance of the <see cref="TripleDesCipher"/> class.
         /// </summary>
         /// <param name="key">The key.</param>
+        /// <param name="iv">The IV.</param>
         /// <param name="mode">The mode.</param>
-        /// <param name="padding">The padding.</param>
+        /// <param name="pkcs7Padding">Enable PKCS7 padding.</param>
         /// <exception cref="ArgumentNullException"><paramref name="key"/> is <see langword="null"/>.</exception>
-        public TripleDesCipher(byte[] key, CipherMode mode, CipherPadding padding)
```

Comment:
if we are not going to support CTR mode, then we might as well pass in System.Security.Cryptography.CipherMode and avoid renaming AesCipherMode